src/main.rs

```diff
@@ -353,71 +55,47 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
 			scenario,
 			send_threshold,
 			remark,
-		} => {
-			let recipe = remark.map_or_else(
-				|| TransactionRecipe::transfer(),
-				|size| TransactionRecipe::remark(size),
-			);
-			match chain {
-				ChainType::Fake => {
-					let sink = FakeTransactionSink::new();
-					let builder = FakeTransactionBuilder::new();
-					execute_scenario(
-						sink,
-						builder,
-						scenario,
-						*unwatched,
-						*send_threshold as usize,
-						&recipe,
-					)
-					.await;
-				},
-				ChainType::Eth => {
-					let transaction_monitor =
-						if *block_monitor { Some(BlockMonitor::new(ws).await) } else { None };
+			tip,
+		} => match chain {
+			ChainType::Fake => {
+				unimplemented!()
+			},
+			ChainType::Eth => {
+				let mut scenario_builder = ScenarioBuilder::new()
+					.with_rpc_uri(ws.to_string())
+					.with_chain_type(chain.clone())
+					.with_send_threshold(*send_threshold as usize)
+					.with_block_monitoring(*block_monitor)
+					.with_watched_txs(!unwatched)
+					.with_tip(*tip);
```

Comment:
tip could be owevritten by lines 74-75 (remark recipe setting).
Maybe tip should be a dedicated field in `ScenarioBuilder` - otherwise it could be overwritten when recipe is set after. (or should be moved from old recipe when new is set).

Reply:
Good catch! Made `with_tip` update its own field and the recipe, while setting the recipe uses the previously set `tip` (0 if not set, and the recipe will be updated once a new tip is set).